amoro-ams/src/main/java/org/apache/amoro/server/catalog/CatalogBuilder.java

```diff
@@ -90,9 +90,11 @@ public static ServerCatalog buildServerCatalog(
       case CATALOG_TYPE_HIVE:
         String amsUri = getAmsURI(serverConfiguration);
         catalogMeta.getCatalogProperties().put(CatalogMetaProperties.AMS_URI, amsUri);
-        catalogMeta
-            .getCatalogProperties()
-            .put(CLIENT_POOL_SIZE, String.valueOf(CLIENT_POOL_SIZE_DEFAULT));
+        if (!catalogMeta.getCatalogProperties().containsKey(CLIENT_POOL_SIZE)) {
```

Comment:
If the logic here overrides the value set, does that mean we can safely remove this logic here?

Reply:
can we add some tests to cover this logic?